# Preprocess for Model Cross-Validation

In [1]:
import os
import numpy as np

In [2]:
PATH_TO_EXP = (
    "/Users/sijianfan/Documents/projects/BiSSGL/datasets/realAnalysis/cdataset"
)
PATH_DATA = os.path.join(PATH_TO_EXP, "cv_data")
if not os.path.isdir(PATH_DATA):
    os.mkdir(PATH_DATA)

In [3]:
import pandas as pd

df_Y = pd.read_table(os.path.join(PATH_TO_EXP, "c_admat_dgc.txt"), index_col=0)
print(df_Y.shape)

(409, 658)


In [5]:
df_Y.head()

,DB00014,DB00035,DB00091,DB00104,DB00115,DB00122,DB00125,DB00126,DB00131,DB00136,...,DB08801,DB08802,DB08804,DB08820,DB08824,DB08835,DB08896,DB08901,DB08906,DB08907
D102100,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D102300,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D102400,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D102500,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D103100,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


Data info:

- 409 diseases $\times$ 658 drugs, binary matrix for drug resistance. 

- We save as 658 drugs $\times$ 409 diseases as $\mathbf{Y}$ to be consistant as DRIMC. 

<!-- - Not all strains are mapped with Assembly ID, so we need to remove the strains first.  -->

<!-- - The final dataset contains: 6949 strains $\times$ 13 drugs, with 9684 SNPs and 33 chemical properties.  -->

In [7]:
drugs_features = pd.read_csv(
    os.path.join(PATH_TO_EXP, "drugs_features.csv"), index_col=0
)
drugs_features.shape

(658, 4020)

In [9]:
drugs_features.head()

,ECFP_0,ECFP_1,ECFP_2,ECFP_3,ECFP_4,ECFP_5,ECFP_6,ECFP_7,ECFP_8,ECFP_9,...,GO_CC_GO:1902711,GO_CC_GO:1902937,GO_CC_GO:1903561,GO_CC_GO:1904115,GO_CC_GO:1904602,GO_CC_GO:1904724,GO_CC_GO:1904813,GO_CC_GO:1990454,GO_CC_GO:1990635,GO_CC_GO:1990904
DB00014,0,1,0,0,1,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00035,0,1,0,0,1,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00091,0,1,0,0,0,1,0,0,0,0,...,0,0,0,0,0,0,1,0,0,0
DB00104,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00115,1,1,0,0,0,0,0,0,0,1,...,0,0,0,0,0,0,0,0,0,0


In [10]:
Y = df_Y.T.copy()
Y.shape

(658, 409)

In [11]:
Y.head()

,D102100,D102300,D102400,D102500,D103100,D103230,D103285,D103780,D104130,D104300,...,D608232,D608266,D608320,D608437,D608456,D608583,D608622,D608636,D608895,D608907
DB00014,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00035,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00091,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00104,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
DB00115,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [12]:
U = np.array(drugs_features, dtype=int)
U.shape

(658, 4020)

In [13]:
diseases_features = pd.read_csv(
    os.path.join(PATH_TO_EXP, "diseases_features.csv"), index_col=0
)
diseases_features.shape

(409, 1860)

In [14]:
diseases_features.head()

,GENE_100188832,GENE_100188863,GENE_100312954,GENE_100499532,GENE_100528023,GENE_100529151,GENE_100861529,GENE_100887743,GENE_100918,GENE_100996949,...,HPO_HP:5200241,HPO_HP:5200263,HPO_HP:5200401,HPO_HP:5200423,HPO_HP:5200430,HPO_HP:5201015,HPO_HP:6000531,HPO_HP:6000768,HPO_HP:6001380,HPO_HP:6001426
D102100,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D102300,0,0,0,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,0
D102400,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
D102500,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
D103100,0,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [16]:
V = np.array(diseases_features, dtype=int)
V.shape

(409, 1860)

In [17]:
from scipy.sparse import coo_matrix

In [19]:
U = coo_matrix(U).tocsr()
V = coo_matrix(V).tocsr()
Y = np.array(Y, dtype=int)
Y = 2 * Y - 1

In [20]:
filename_staged = os.path.join(PATH_DATA, "staged_dataset.gz")

import gzip
import pickle

with gzip.open(filename_staged, "wb+", 4) as fout:
    pickle.dump((U, V, Y), fout)